src/sys/socket/consts.rs

```diff
@@ -203,25 +196,22 @@ mod os {
     pub const SO_RESTRICT_DENYSET: c_int    = 0x80000000;
 
     // Socket options for TCP sockets
-    pub const TCP_NODELAY: c_int = 1;
-    pub const TCP_MAXSEG: c_int = 2;
-    #[cfg(any(target_os = "macos",
-              target_os = "ios"))]
+    pub const TCP_NODELAY: c_int = libc::TCP_NODELAY;
+    pub const TCP_MAXSEG: c_int = libc::TCP_MAXSEG;
+    #[cfg(any(target_os = "macos", target_os = "ios"))]
     pub const TCP_KEEPALIVE: c_int = libc::TCP_KEEPALIVE;
-    #[cfg(target_os = "freebsd")]
+    #[cfg(target_os = "freebsd", target_os = "netbsd")]
```

Comment:
Missing an `any(` expression in your `#[cfg`.  Also, libc has not defined `TCP_KEEPIDLE` for NetBSD.  Only for FreeBSD, Dragonfly, Linux, Android, and Solaris.